lua/nvim-treesitter/languagetree.lua

```diff
@@ -0,0 +1,153 @@
+local parsers = require'nvim-treesitter.parsers'
+local queries = require'nvim-treesitter.query'
+local tsutils = require'nvim-treesitter.ts_utils'
+local TSHighlighter = require'vim.treesitter.highlighter'
+
+local ns = vim.api.nvim_create_namespace("treesitter/highlighter")
+
+local LanguageTree = {}
+LanguageTree.__index = LanguageTree
+
+local trees = { }
+
+function LanguageTree.new(bufnr, lang, not_root)
+  local buf
+  if not bufnr or bufnr == 0 then
+    buf = vim.api.nvim_get_current_buf()
+  else
+    buf = bufnr
+  end
+
+  local parser = parsers.get_parser(buf, parsers.ft_to_lang(lang))
+  if not parser then return end
+
+  local query = queries.get_query(lang, "highlights")
+  if not query then return end
+
+  local self = setmetatable({
+    highlighter=TSHighlighter.new(parser, query),
+    parser=parser,
+    children={}}, LanguageTree)
```

Comment:
new line after `}` maybe ? this formatting looks weird

Reply:
Reformatted it, should be better now

Reply:
I was thinking about writing a formatter for lua based on treesitter (again haha). Maybe when i have some more free time.